contract/lib/rebase/tzprofiles.ts

```diff
@@ -0,0 +1,209 @@
+import * as taquito from '@taquito/taquito';
+import * as tzip16 from '@taquito/tzip16';
+import type { BeaconWallet } from '@taquito/beacon-wallet';
+import type {NetworkType} from '@airgap/beacon-sdk';
+
+import * as v0Auth from './control/v0/provider';
+import * as v1Auth from './control/v1/provider';
+
+import { default as RebaseControlClient } from 'rebase/lib/hashref/client/ControlClient';
+import { default as RebaseViewClient } from 'rebase/lib/hashref/client/ViewClient';
+
+import {
+  Content, Hash, Ref, Location, fromRawStorage, screener,
+} from './storage';
+
+import hasher from './hasher/hasher';
+
+import { ByOwner } from 'rebase/lib/hashref/storage/hash/tezos/view/provider';
+import { HashControl, HashViewer } from 'rebase/lib/hashref/client/provider/hashStorage';
+import { ByOwnerHashControl, locationOptsFromNetwork } from 'rebase/lib/hashref/storage/hash/tezos/control/provider';
+import {
+  contractAddressesFromOwner, loadSigner, Signer, TQContract,
+} from 'rebase/lib/hashref/storage/hash/tezos/tezos';
+import { toHashViewer } from './view/v0/provider';
+
+import { ContentViewer, ContentControl } from 'rebase/lib/hashref/client/provider/contentStorage';
+import { contentViewer } from 'rebase/lib/hashref/storage/content/kepler/view/provider';
+import { contentControl } from 'rebase/lib/hashref/storage/content/kepler/control/tezos/provider';
+import { ControlOpts } from 'rebase/lib/hashref/client/MultiClient';
+
+import {Entry as RawEntry} from 'rebase/lib/hashref/client/abstraction/storage';
+
+export type Entry = RawEntry<Content, Hash, Ref>;
+export type ViewClient = RebaseViewClient<Content, Hash, Location, Ref>;
+export class ControlClient extends RebaseControlClient<Content, Hash, Location, Ref> {
+  constructor(
+    hasher: (c: Content) => Promise<Hash>,
+    contentControl: ContentControl<Content, Ref>,
+    hashControl: ByOwnerHashControl<Hash, Ref>,
+  ) {
+    super(
+      hasher,
+      contentControl,
+      hashControl
+    );
+  }
+
+  getAddr() {
+    let x = this.hashControl as ByOwnerHashControl<Hash, Ref>;
+    return x.contractAddr();
+  }
+}
+
+const metadataUrl = 'https://tzprofiles.com/tzip016_metadata.json';
+
+export function makeContentViewer(domain: string): ContentViewer<Content, Ref> {
+  const f = contentViewer(domain);
+  return async (ref: Ref) => JSON.parse(await f(ref)) as Content;
+}
+
+export function makeHashViewer(searchAddress: string): HashViewer<Hash, Location, Ref> {
+  return toHashViewer(searchAddress, 'GetClaims', fromRawStorage, screener);
+}
+
+export type ViewClientOpts = {
+  searchAddress: string;
+  keplerDomain: string;
+  network: NetworkType;
+};
+
+export function makeViewClient(opts: ViewClientOpts): ViewClient {
+  return new RebaseViewClient(
+    hasher,
+    makeContentViewer(opts.keplerDomain),
+    makeHashViewer(opts.searchAddress),
+  );
+}
+
+export function viewerOpts(network: NetworkType, ownerAddress: string): ByOwner {
```

Comment:
I'm a bit confused between the viewer that uses the indexer, and this viewer which does the full resolution?